cachi2/core/package_managers/yarn_classic/project.py

```diff
@@ -0,0 +1,171 @@
+"""
+Parse the relevant files of a yarn project.
+
+It also provides basic utility functions. The main logic to resolve and prefetch
+the dependencies should be implemented in other modules.
+"""
+
+import json
+import logging
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Any, Literal, Union
+
+from pyarn import lockfile  # type: ignore
+
+from cachi2.core.errors import PackageRejected
+from cachi2.core.rooted_path import RootedPath
+
+log = logging.getLogger(name=__name__)
+
+ConfigKind = Literal["package_json", "yarnlock"]
+
+
+@dataclass
+class _CommonConfigFile(ABC):
+    """A base class for representing a config file.
+
+    :param path: the path to the config file, relative to the request source dir
+    :param data: the raw data for the config file content
+    """
+
+    _path: RootedPath
+    _data: dict[str, Any]
+
+    @property
+    def data(self) -> dict[str, Any]:
+        return self._data
+
+    @property
+    def path(self) -> RootedPath:
+        return self._path
+
+    @property
+    @abstractmethod
+    def config_kind(self) -> ConfigKind:
+        """Return kind of ConfigFile instance."""
+        pass
```

Comment:
Non-blocking: `pass` is redundant here (since the method already has one statement).